### PR TITLE
Set feature flags in your halconfig

### PR DESCRIPTION
**Pull request by lwander**
features

Right now these correspond to what's configurable in deck outside of netflix-only features.

@duftler or @jtk54 or @danielpeach PTAL

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -78,6 +78,21 @@ To delete an account, run:
 $ hal config provider PROVIDER delete-account ACCOUNT-NAME
 ```
 
+#### hal config features
+
+To enable either the experimental or non-standard features of Spinnaker, use
+the command:
+
+```
+$ hal config features [--features true/false]
```
Comment (duftler) [inline README.md]: Should this be `--feature`?

File: halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/services/v1/Daemon.java [hunk 2]
```diff
@@ -30,11 +31,20 @@ public static String getCurrentDeployment() {
...
+  public static void setFeatures(String deploymentName, boolean validate, Features features) {
+    ResponseUnwrapper.get(service.setFeatures(deploymentName, validate, features));
+  }
+
   public static Account getAccount(String deploymentName, String providerName, String accountName, boolean validate) {
     Object rawAccount = ResponseUnwrapper.get(getService().getAccount(deploymentName, providerName, accountName, validate));
     return getObjectMapper().convertValue(rawAccount, Providers.translateAccountType(providerName));
   }
 
+
```
Comment (duftler) [inline halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/services/v1/Daemon.java]: Extraneous blank line.

File: halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/services/v1/FeaturesService.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    filter = filter.withAnyHalconfigFile().setConfigNode("features");
+
+    List<Features> matching = lookupService.getMatchingNodesOfType(filter, Features.class)
+        .stream()
+        .map(n -> (Features) n)
+        .collect(Collectors.toList());
+
+    switch (matching.size()) {
+      case 0:
+        throw new RuntimeException("No features could be found. This is a bug.");
```
Comment (duftler) [inline halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/services/v1/FeaturesService.java]: s/could be/were

File: halyard-web/src/main/java/com/netflix/spinnaker/halyard/controllers/v1/FeaturesController.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    return builder.build();
+  }
+
+  @RequestMapping(value = "/", method = RequestMethod.PUT)
+  DaemonResponse<Void> setFeatures(
+      @PathVariable String deployment,
+      @RequestParam(required = false, defaultValue = DefaultControllerValues.validate) boolean validate,
+      @RequestParam(required = false, defaultValue = DefaultControllerValues.severity) Severity severity,
+      @RequestBody Object rawFeatures) {
+    NodeFilter filter = new NodeFilter()
```
Comment (duftler) [inline halyard-web/src/main/java/com/netflix/spinnaker/halyard/controllers/v1/FeaturesController.java]: Looks like `.setDeployment()` will fit on the same line.

File: halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/services/v1/FeaturesService.java (hunk 3 above)
Comment (lwander) [inline halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/services/v1/FeaturesService.java]: Just curious - why this change?

**Comment by duftler** [comment]
Just sounded a bit funny to me. And less words better if it means the same
thing.

**Comment by lwander** [comment]
SGTM
